src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs

```diff
@@ -333,21 +334,23 @@ public void InitializeService(IProtocolEndpoint serviceHost)
                             sqlConn.Dispose();
                         }
                     };
+                    response.Result = true;
                 }
-                catch
+                else
                 {
-                    // Ensure that the connection is closed if any error occurs while starting up the task
-                    sqlConn.Dispose();
-                    throw;
+                    response.Result = false;
                 }
 
+                await requestContext.SendResult(response);
             }
-            else
+            catch (Exception e)
             {
-                response.Result = false;
+                await requestContext.SendError(e);
+                if (sqlConn != null)
+                {
+                    sqlConn.Dispose();
```

Comment:
What about if the task starts up successfully but then an error occurs during execution? Are we guaranteed to have StatusChanged with IsCompleted = true called in that case?

Reply:
Looking at the SqlTask class, IsCompleted only gets set to false if the status is InProgress or NotStarted. On other results like Failed or Canceled it gets set to true.

Reply:
https://github.com/microsoft/sqltoolsservice/blob/main/src/Microsoft.SqlTools.ServiceLayer/TaskServices/SqlTask.cs#L355